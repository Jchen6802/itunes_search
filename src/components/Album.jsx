import PropTypes from 'prop-types';
import styled from 'styled-components';
import React, { Component} from 'react';

const AlbumCard = styled.div`
    width: 100%;
    height: 100%;
    border: 1px solid black;
    padding: 5px;
`;

const AlbumName = styled.div`
    margin-top: 5px;
    font-size: .9em;
    font-weight: bold;
`;

const ArtistName = styled.div`
    font-size: .7em;
`;

class Album extends Component {
    render() {
        return <div className="column">
            <AlbumCard>
                <img
                    alt={this.props.albumName}
                    src={this.props.albumCover}/>
                <AlbumName>{this.props.albumName}</AlbumName>
                <ArtistName>{this.props.artistName}</ArtistName>
            </AlbumCard>
        </div>;
    }
}

Album.propTypes = {
    albumName: PropTypes.string.isRequired,
    albumCover: PropTypes.string.isRequired,
    artistName: PropTypes.string.isRequired,
};

export default Album;
