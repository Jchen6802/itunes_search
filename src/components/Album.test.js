import React from 'react';
import renderer from 'react-test-renderer';

import Album from './Album.jsx';

it('will render as expected', () => {
    const props = {
        albumName: 'someAlbum',
        albumCover: 'someAlbumCoverUrl',
        artistName: 'someArtist'
    };

    const album = renderer.create(
        <Album {...props}/>
    ).toJSON();

    expect(album).toMatchSnapshot();
});
