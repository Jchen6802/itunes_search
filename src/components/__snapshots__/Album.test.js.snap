// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will render as expected 1`] = `
<div
  className="column"
>
  <div
    className="sc-bdVaJa gfWgdN"
  >
    <img
      alt="someAlbum"
      src="someAlbumCoverUrl"
    />
    <div
      className="sc-bwzfXH hTbHfO"
    >
      someAlbum
    </div>
    <div
      className="sc-htpNat fpkfan"
    >
      someArtist
    </div>
  </div>
</div>
`;
